### PR TITLE
[#859] Improve logging message when repo (branch) isn't found in repo-config.csv

### LINKED ISSUE #859: Improve logging message when repo (branch) isn't found in repo-config.csv
Given this config setup (note that `testrepo-Beta (master)` doesn't exist in repo-config.csv but is included in author-config.csv)
image

The log message `[WARNING] Repository https://github.com/reposense/testrepo-Beta.git is not found in repo-config.csv.` doesn't include which branch of testrepo-Beta isn't found in the repo-config.csv
image

### PR DESCRIPTION
**Pull request by 0blivious**
Fixes #859 
```
The message '[WARNING] Repository SOME_REPO is not found in
repo-config.csv.' appears when a corresponding configuration for
`author-config` cannot be found in `repo-config`. 

However, it does not include the branch name, of the SOME_REPO, that
cannot be found in the repo-config.csv. This can confuse the users as
they may not know the exact error since there can be multiple lines of
configuration targeting different branches of one repository.

Let's include the branch name in the warning message to provide users
better clarity of the error.
```

### CODE REVIEW

**Comment by jamessspanggg** [comment]
@0blivious please wrap the commit message body at 72 characters, the conventions can be referred [here](https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message)

File: src/main/java/reposense/model/RepoConfiguration.java [hunk 1]
```diff
@@ -96,7 +96,8 @@ public static void merge(List<RepoConfiguration> repoConfigs, List<AuthorConfigu
 
             if (matchingRepoConfig == null) {
                 logger.warning(String.format(
-                        "Repository %s is not found in repo-config.csv.", authorConfig.getLocation()));
+                        "Repository %s (branch %s) is not found in repo-config.csv.",
+                        authorConfig.getLocation(), authorConfig.getBranch()));
```
Comment (eugenepeh) [inline src/main/java/reposense/model/RepoConfiguration.java]: Sorry, I just remembered that the branch is an optional field in author config. 
Can you tweak this message, such that it wont print the branch name if the field is using default branch?
This is to prepare the structure for https://github.com/reposense/RepoSense/issues/865

Comment (0blivious) [reply src/main/java/reposense/model/RepoConfiguration.java]: Resolved

File: src/main/java/reposense/model/AuthorConfiguration.java [hunk 2]
```diff
@@ -215,4 +215,8 @@ public Author getAuthor(String name, String email) {
         }
         return Author.UNKNOWN_AUTHOR;
     }
+
+    public boolean isUsingDefaultBranch() {
```
Comment (eugenepeh) [inline src/main/java/reposense/model/AuthorConfiguration.java]: `isDefaultBranch` should be good enough

File: src/main/java/reposense/model/RepoConfiguration.java [hunk 3]
```diff
@@ -95,8 +95,15 @@ public static void merge(List<RepoConfiguration> repoConfigs, List<AuthorConfigu
             RepoConfiguration matchingRepoConfig = getMatchingRepoConfig(repoConfigs, authorConfig);
 
             if (matchingRepoConfig == null) {
-                logger.warning(String.format(
-                        "Repository %s is not found in repo-config.csv.", authorConfig.getLocation()));
+                if (authorConfig.isUsingDefaultBranch()) {
+                    logger.warning(String.format(
+                            "Repository %s is not found in repo-config.csv.",
+                            authorConfig.getLocation(), authorConfig.getBranch()));
```
Comment (eugenepeh) [inline src/main/java/reposense/model/RepoConfiguration.java]: perhaps, let's do something like this:
```
String branchInfo = authorConfig.isDefaultBranch() 
        ? String.format(" (branch %s)",0 authorConfig.getBranch()) 
        : "";
logger.warning(String.format(
        "Repository %s %s is not found in repo-config.csv.",
        authorConfig.getLocation(), branchInfo));
```
Generally try to use lesser long strings as it affects the size of compiled output.

Comment (0blivious) [reply src/main/java/reposense/model/RepoConfiguration.java]: Fixed

**Comment by 0blivious** [comment]
Comment message updated.

**Pull request merged**